mama/c_cpp/src/c/clientmanageresponder.c

```diff
@@ -90,231 +90,234 @@ static void commandDestroyCB (wList list, void *element, void *closure)
-					{
-						ret = pd;
-					}
-				}
-			}
-
-			/* Free the array of publishers. */
-			free(impl->publishers);
+    if(impl != NULL)
+    {        
+        /* Publishers and subscribers are held for each sub transport bridge, each must be freed in
+         * turn before the corresponding arrays can be deleted.
+         * Note that these are enumerate separately in case something went wrong during creation and
+         * one array wasn't created successfully.
+         */
+        ret = MAMA_STATUS_OK;
+
+        /* Free the subscriptions. */
+        if(impl->subscriptions != NULL)
+        {
+            /* Enumerate all subscribers. */
+            mamaSubscription *nextSubscription    = impl->subscriptions;
+            int nextBridgeIndex                    = 0;
+            for(; nextBridgeIndex<impl->numberTransportBridges; nextBridgeIndex++, nextSubscription++)
+            {
+                /* Free the subscription. */
+                if(*nextSubscription != NULL)
+                {
+                    /* Destroy the subscription, note that the first failure return code will be preserved.
+                     */
+                    mama_status sd = mamaSubscription_destroyEx(*nextSubscription);
+                    if(ret == MAMA_STATUS_OK)
+                    {
+                        ret = sd;
+                    }
+                }
+            }
+
+            /* Free the array itself. */
+            free(impl->subscriptions);
+            impl->subscriptions = NULL;
+        }
+
+        /* Free the publishers. */
+        if(impl->publishers != NULL)
+        {
+            /* Enumerate all publishers. */
+            mamaPublisher *nextPublisher    = impl->publishers;            
+            int nextBridgeIndex                = 0;
+            for(; nextBridgeIndex<impl->numberTransportBridges; nextBridgeIndex++, nextPublisher++)
+            {                
+                /* Free the publisher, note that the first failure return code will be preserved. */
+                if(*nextPublisher != NULL)
+                {
+                    mama_status pd = mamaPublisher_destroy(*nextPublisher);
+                    if(ret == MAMA_STATUS_OK)
+                    {
+                        ret = pd;
+                    }
+                }
+            }
+
+            /* Free the array of publishers. */
+            free(impl->publishers);
             impl->publishers = NULL;
-		}
+        }
 
-		/* Destroy all pending commands. */
-		if(impl->mPendingCommands != NULL)
-		{
-			list_destroy(impl->mPendingCommands, commandDestroyCB, impl);
-		}
+        /* Destroy all pending commands. */
+        if(impl->mPendingCommands != NULL)
+        {
+            list_destroy(impl->mPendingCommands, commandDestroyCB, impl);
+        }
 
-		/* Free the impl itself. */
-		free(impl);
-	}
+        /* Free the impl itself. */
+        free(impl);
+    }
 
-	return ret;
+    return ret;
 }
 
```

Comment:
There were lots of tabs in this file, which got changed to 4 spaces when I did global search/replace. I restored the tab on this line which should make the indent revert to original.